sdk/iot/common/CMakeLists.txt

```diff
@@ -11,20 +11,22 @@ set(CMAKE_C_STANDARD 99)
 
 include(CheckAndIncludeCodeCov)
 
-add_library (
-    ${TARGET_NAME}
-    src/az_iot_common.c
+add_library (${TARGET_NAME}
+  src/az_iot_common.c
 )
 
 target_include_directories (${TARGET_NAME} PUBLIC inc)
 target_include_directories (${TARGET_NAME} PRIVATE src)
 
-target_link_libraries(${TARGET_NAME} PRIVATE az_core)
+target_link_libraries(${TARGET_NAME}
+  PUBLIC
+    az::core
+)
```

Comment:
If I understand correctly, changing this from private to public is specifically what is fixing https://github.com/Azure/azure-sdk-for-c/issues/824. Correct?

Reply:
Yes exactly